Library/src/Systems/Quota/Quota.cpp

```diff
@@ -104,6 +136,38 @@ void FeatureLimitResult::OnResponse(const csp::services::ApiResponseBase* ApiRes
 	}
 }
 
+int FeatureLimitResult::ParseErrorCode(const csp::common::String& Value)
+{
+	if (Value == "group_spaceownerquota")
+		return (int) EQuotaResultFailureReason::SpaceOwnerQuota;
+
+	if (Value == "scopes_concurrentusersquota")
+		return (int) EQuotaResultFailureReason::ConcurrentUsersQuota;
+
+	if (Value == "assetdetail_objectcapturequota")
+		return (int) EQuotaResultFailureReason::ObjectCaptureQuota;
+
+	if (Value == "assetdetail_audiovideoquota")
+		return (int) EQuotaResultFailureReason::AudioVideoQuota;
+
+	if (Value == "assetdetail_totaluploadsizeinkilobytes")
+		return (int) EQuotaResultFailureReason::TotalUploadSizeInKilobytes;
+
+	if (Value == "agoraoperation_groupownerquota")
+		return (int) EQuotaResultFailureReason::AgoraOwnerQuota;
+
+	if (Value == "openaioperation_userquota")
+		return (int) EQuotaResultFailureReason::OpenAIUserQuota;
+
+	if (Value == "shopify_userquota")
```

Comment:
With errors like these that can be returned for a variety of requests, is there a way we could have only one place that parses them, to avoid the duplicate code?

If there's a good reason for the duplicate code, I could be perusaded to keep it as-is.

Reply:
sorted

Reply:
The duplicate code is useful because each enum is unique to the result. I can't think of a way we can guarantee developers will not use conflicting enum values.